Directory.Build.props

```diff
@@ -6,6 +6,18 @@
         <SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>
     </PropertyGroup>
 
+    <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('Linux'))">
+        <DefineConstants>_LINUX</DefineConstants>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('OSX'))">
+        <DefineConstants>_MACOS</DefineConstants>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
+        <DefineConstants>_WINDOWS</DefineConstants>
+    </PropertyGroup>
+
```

Comment:
Inspired by https://stackoverflow.com/a/65208710/227779 and https://blog.magnusmontin.net/2018/11/05/platform-conditional-compilation-in-net-core/, this approach seems to be a quite nice way to handle platform-specific code. :tada: